### PR TITLE
Add a threshold parameter for prediction

### PR DESCRIPTION
**Pull request by maudetes**
Solving #7. We now have a threshold parameter for the classifier that will set a minimum confidence score to return a tag. We also added the `threshold` parameter on the Rest Server `/intention/` route.
By default, the threshold value is 0.

### CODE REVIEW

File: include/katanoisi/classifier.h [hunk 1]
```diff
@@ -14,7 +14,7 @@ class classifier {
   }
 
   std::vector<std::pair<fasttext::real, std::string>>
-  prediction(std::__cxx11::string &text, int count);
+  prediction(std::__cxx11::string &text, int count, float threshold=0.0);
```
Comment (NoelM) [inline include/katanoisi/classifier.h:17]: Why a `float` not a `double`?

File: src/rest_server.cpp [hunk 2]
```diff
@@ -85,10 +85,14 @@ void rest_server::doClassificationPost(const Rest::Request &request,
   response.headers().add<Http::Header::AccessControlAllowOrigin>("*");
   nlohmann::json j = nlohmann::json::parse(request.body());
   int count = 10;
+  float threshold = 0.0;
```
Comment (NoelM) [inline src/rest_server.cpp:88]: Could be a member of the class `classifier`?

File: include/katanoisi/rest_server.h [hunk 3]
```diff
@@ -49,7 +49,7 @@ class rest_server {
                             Http::ResponseWriter response);
 
   std::vector<std::pair<fasttext::real, std::string>>
-  askClassification(std::string &text, std::string &domain, int count);
+  askClassification(std::string &text, std::string &domain, int count, float threshold);
```
Comment (NoelM) [inline include/katanoisi/rest_server.h:52]: Why not passed by reference?

File: src/rest_server.cpp (hunk 2 above)
Comment (maudetes) [reply src/rest_server.cpp:88]: The threshold can change at each classification called, it doesn't seem to make sense to make it a `classifier` class member.

File: include/katanoisi/classifier.h (hunk 1 above)
Comment (maudetes) [reply include/katanoisi/classifier.h:17]: Changed in next commit

File: include/katanoisi/rest_server.h (hunk 3 above)
Comment (maudetes) [reply include/katanoisi/rest_server.h:52]: Changed in next commit

File: include/katanoisi/classifier.h (hunk 1 above)
Comment (maudetes) [reply include/katanoisi/classifier.h:17]: Actually, FastText expects a `real` value for the threshold parameter, that they defined as a `float`.

File: include/katanoisi/rest_server.h (hunk 3 above)
Comment (maudetes) [reply include/katanoisi/rest_server.h:52]: If passed by reference, we can't set a default right value, except if const which is not the case in fastText.

**Comment by NoelM** [comment]
LGTM

**Pull request merged**